specs/release.md

```diff
@@ -0,0 +1,319 @@
+
+Every commit merged to `main` MUST automatically deploy to the development environment. This deployment occurs after all CI checks pass and MUST NOT require manual approval.
+
+The development deployment workflow MUST:
+
+1. Verify all CI checks passed (tests, type checking, coverage)
+2. Deploy to Cloudflare Workers development environment
+3. Create a GitHub Deployment record with environment "development"
+4. Update deployment status to "success" or "failure"
+5. Verify deployment with health check endpoint
+6. Rollback automatically if health check fails
+
+Development deployments MUST complete within 3 minutes of merge to `main`. Deployment failures MUST NOT block subsequent development deployments (fail forward, not fail stop).
+
+### Production Deployment
+
+Production deployments MUST be triggered manually via GitHub Actions workflow dispatch. This manual gate provides control over production releases while maintaining automation for everything except the final approval decision.
+
+The production deployment workflow MUST:
+
+1. Accept a workflow dispatch event from an authorized maintainer
+2. Verify all CI checks passed on the commit being deployed
+3. Re-run tests as final validation (even if CI already passed)
+4. Deploy to Cloudflare Workers production environment
+5. Create a GitHub Deployment record with environment "production"
+6. Wait for manual approval if GitHub Environment protection rules are configured
+7. Update deployment status to "success" or "failure"
+8. Verify deployment with health check endpoint
+9. Rollback automatically if health check fails
+
+Production deployments SHOULD occur during business hours when maintainers can monitor for issues. Automated rollback handles immediate failures; delayed issues require manual intervention.
+
+### GitHub Deployments API Integration
+
+All deployments (development and production) MUST create GitHub Deployment records via the Deployments API. This provides centralized deployment history, status tracking, and integration with GitHub's environment protection features.
+
+For each deployment, the workflow MUST:
+
+1. Create a deployment record with the commit SHA being deployed
+2. Set the environment name ("development" or "production")
+3. Set deployment status to "in_progress" during deployment
+4. Update status to "success" if deployment and verification succeed
+5. Update status to "failure" if any step fails
+6. Include deployment URL in the status update
+
+GitHub Environment protection rules SHOULD be configured for production to:
+
+- Require manual approval from designated reviewers
+- Enforce deployment branch restrictions (only `main`)
+- Apply wait timer (optional, e.g., 5 minute cooldown between deployments)
+
+## Versioning Strategy
+
+The project DOES NOT use semantic versioning tags for deployments. Version numbers exist for documentation and tracking purposes but do NOT trigger deployments or control which code runs in production.
```

Comment:
So I don't really know what the answer is here but I don't have much interest in managing version numbers. All I really value is being able to identify releases and the code they correspond to. I kinda like the idea of the version number being YEAR.RELEASE.HOTFIX Where year is the last 2 digits of the year the release is planned for(there may be some slight discrepancy here around the turn of the year which is not a big deal), release is the number this release represents in the year and should always be the increment of the previous release. Hotfix is specific nomenclature for when a release is done with the hotfix process.